### ISSUE TITLE
Can't debug on IntelliJ Ultimate

### ISSUE THREAD

**Issue by joeflack4** [hapifhir/hapi-fhir-jpaserver-starter#341]
## Preface
Related issue: #188 . I posted in there, but I couldn't re-open the issue, so I'm making a new issue.

Failed debug configs I tried:
1. [Server 1](https://github.com/HOT-Ecosystem/hapi-fhir-jpaserver-starter), 'boot'
2. [Server 1](https://github.com/HOT-Ecosystem/hapi-fhir-jpaserver-starter), Tomcat
3. [Server 2](https://github.com/pnrobinson/hapi-pheno-server/tree/develop), 'boot'

## Description
I'm trying to debug as described above and in [the readme]([1.txt](https://github.com/hapifhir/hapi-fhir-jpaserver-starter/files/8405840/1.txt)
[2.txt](https://github.com/hapifhir/hapi-fhir-jpaserver-starter/files/8405841/2.txt)). But I get errors.

I'm working with two different clones; (1) is my server, which is basically a fork with almost no changes, and (2) is my colleague's, where he has made some changes.

## Server 1: boot
I have no problem running this server in the command line with `docker-compose` or with `mvn -Djetty.port=8888 jetty:run`.

Repo: https://github.com/HOT-Ecosystem/hapi-fhir-jpaserver-starter

Debugger error full logs: 
[1.txt](https://github.com/hapifhir/hapi-fhir-jpaserver-starter/files/8405879/1.txt)

This is the only error I see:
```
2022-04-03 19:57:46.606 [http-nio-8080-exec-1] ERROR org.thymeleaf.TemplateEngine [TemplateEngine.java:1136] [THYMELEAF][http-nio-8080-exec-1] Exception processing template "home": An error happened during template parsing (template: "ServletContext resource [/WEB-INF/templates/home.html]")
org.thymeleaf.exceptions.TemplateInputException: An error happened during template parsing (template: "ServletContext resource [/WEB-INF/templates/home.html]")
...
Caused by: java.io.FileNotFoundException: Could not open ServletContext resource [/WEB-INF/templates/home.html]
```

I don't have `home.html` in my `WEB-INF/templates/` folder. But even on the current master branch of the main project, i'ts not there either:
<img width="953" alt="Screen Shot 2022-04-03 at 8 07 27 PM" src="https://user-images.githubusercontent.com/13045020/161455077-dcbbe9ca-7c9a-4aa8-9e22-e5a4c1565439.png">

Why is it searching for something that doesn't exist?
How can I fix this error and debug locally?

## Server 1: Tomcat
I tried following the readme instructions for running on tomcat, but no luck.

Repo: https://github.com/HOT-Ecosystem/hapi-fhir-jpaserver-starter

Full logs: [1-tomcat.txt](https://github.com/hapifhir/hapi-fhir-jpaserver-starter/files/8405931/1-tomcat.txt)

I'm not seeing any errors, but nothing opens up in the browser:
<img width="732" alt="Screen Shot 2022-04-03 at 8 24 42 PM" src="https://user-images.githubusercontent.com/13045020/161455826-1c627ede-28b4-405d-9924-5a882eab9473.png">

Here's my debug config. I set it up like in the readme:
<img width="1039" alt="Screen Shot 2022-04-03 at 8 23 21 PM" src="https://user-images.githubusercontent.com/13045020/161455857-96dddd8a-2169-4d11-89f0-9cad35c40df7.png">

<img width="1019" alt="Screen Shot 2022-04-03 at 8 23 38 PM" src="https://user-images.githubusercontent.com/13045020/161455858-bd917fe4-73d4-423b-9405-9a1c1abd41ae.png">

## Server 2: boot
This is my colleague's server. I'm able to run it with `mvn -Djetty.port=8888 jetty:run`. But can't run via debugger either.

Repo: https://github.com/pnrobinson/hapi-pheno-server/tree/develop

Debugger error full logs: [2.txt](https://github.com/hapifhir/hapi-fhir-jpaserver-starter/files/8405841/2.txt)

Error 1:
```
java.io.FileNotFoundException: Could not open ServletContext resource [/WEB-INF/templates/home.html]
```

This is the same error I'm seeing with "Server 1" above.

Error 2:
```
java.lang.NullPointerException: Cannot invoke "ca.uhn.fhir.context.support.ValueSetExpansionOptions.isFailOnMissingCodeSystem()" because "theExpansionOptions" is null
```

I know that this is a problem with some new code we've added to this project, and I think I can figure out how to fix this. But I'm curious as to why I can run the server normally without issue, but only get this error when running with the debugger.

**Comment by joeflack4** [comment]
Closing, as @jkiddo responded to another issue with the quote below. I confirmed that he is correct.
> The UI does not load when you are running it as a debug instance from within Intellij. This is a known bug. All the rest of the FHIR server does run. Go to eg. /fhir/metadata when you are debugging - then you should see that all the REST endpoints are working perfectly fine.

@jkiddo Do you want me to open a new issue for this error that I'm still seeing (even though the server runs anyway)?:
```
java.io.FileNotFoundException: Could not open ServletContext resource [/WEB-INF/templates/home.html]
```

**Comment by jkiddo** [comment]
@joeflack4 the issue is already listed here: https://github.com/hapifhir/hapi-fhir-jpaserver-starter/issues/165

**Comment by joeflack4** [comment]
@jkiddo That's not the error I was having. I had no problem running that command; never got the error mentioned in #165. This was related to running the debugger in 2 specific ways mentioned in the the documentation, and different resulting errors.

**Comment by jkiddo** [comment]
@joeflack4 - looking at your errors:

* https://github.com/hapifhir/hapi-fhir-jpaserver-starter/files/8405840/1.txt : `Web server failed to start. Port 8080 was already in use.` - that is caused as you have something else running on 8080 already
* https://github.com/hapifhir/hapi-fhir-jpaserver-starter/files/8405841/2.txt : `ERROR c.u.f.jpa.term.BaseTermReadSvcImpl [BaseTermReadSvcImpl.java:1796] Failed to pre-expand ValueSet: Cannot invoke "ca.uhn.fhir.context.support.ValueSetExpansionOptions.isFailOnMissingCodeSystem()" because "theExpansionOptions" is null` : it looks like you introduced a bug that blocks the service entirely from starting up
* https://github.com/hapifhir/hapi-fhir-jpaserver-starter/files/8405879/1.txt : the log shows that the server is actually started but fails to render the html files - which is the case of https://github.com/hapifhir/hapi-fhir-jpaserver-starter/issues/165

**Comment by joeflack4** [comment]
Hey again. Yeah, the port 8080 being used already, I saw and fixed that quickly. And I deleted and re-uploaded `1.txt`, but maybe it used the old one that I uploaded. That was no longer a problem. The problem I was still having was the 3rd one you mentioned; the `home.html` template not found / failing to render. I am aware of the second issue as well, and I mentioned that earlier in the OP.

So the real only remaining issue is the 3rd one, the `home.html` one. So I guess #165 is relevant in that regard. But this doesn't happen when I run the server normally. It only happens when I try to enter debug mode from IntelliJ.

In any case, it's actually a non-blocking error, and as pointed out before, I can still debug. It's just that I can't use the UI. Which isn't a major problem for my development / debugging. So therefore I think it's OK to keep this issue closed. Thanks for looking into this.
